drivers/timer/npcx_itim_timer.c

```diff
@@ -97,15 +100,22 @@ static inline uint64_t npcx_itim_get_sys_cyc64(void)
 
 static inline void npcx_itim_evt_enable(void)
 {
+	uint64_t cyc_start;
+
 	/* Enable event timer and wait for it to take effect */
 	evt_tmr->ITCTS32 |= BIT(NPCX_ITCTSXX_ITEN);
 
 	/*
 	 * Usually, it need one clock (30.5 us) to take effect since
 	 * asynchronization between core and itim32's source clock (LFCLK).
 	 */
+	cyc_start = npcx_itim_get_sys_cyc64();
 	while (!IS_BIT_SET(evt_tmr->ITCTS32, NPCX_ITCTSXX_ITEN)) {
-		continue;
+		if (npcx_itim_get_sys_cyc64() - cyc_start >
+						NPCX_ITIM_EN_TIMEOUT_CYCLES) {
+			LOG_ERR("Timeout: enable EVT timer!");
```

Comment:
How is this reported? Should this function return an error?

Reply:
I have thought about it. During this function is called, the callers in this driver needn't report error code to the upper layer. 
It was called for the first time in [npcx_itim_start_evt_tmr_by_tick](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/timer/npcx_itim_timer.c#L146) function. And the caller, [z_clock_set_timeout](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/timer/npcx_itim_timer.c#L215), of this function is a nonvalue-returning api function, Then it was called in [ISR](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/timer/npcx_itim_timer.c#L172) and the caller is a nonvalue-returning function, too.

That's why I put an error message here to inform something wrong. (Although the enable bit, ITEN, takes effect after one clock cycle if ITIM's source clock is 32KHz is hardware-proven, eliminating the unbounded loops possibly is the target of this PR) In case this function is called somewhere else, I will add reporting error code later.

Reply:
Done.